File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& arr, int k) {
        long long i=0,j=0;
        double sum=0;
        double maxy=INT_MIN;
        int N=arr.size();
       while(j<N)
       {
            sum+=arr[j];
            if(j-i+1<k)
                    j++;
            else if(j-i+1==k)
            {
            
                maxy=max(maxy,sum);
                sum-=arr[i];
                j++;
                i++;
            }
    
       }    cout<<maxy<<" ";
          return maxy/k;
    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int n = nums.length;
        double sum = 0;
        double ans = Integer.MIN_VALUE;
        int i = 0;
        int j = 0;
        while (j < n) {
            sum += nums[j];
            if (j - i + 1 == k) {
                ans = Math.max(ans, sum / (double) k);
                sum -= nums[i];
                i++;
            }
            j++;
        }
        return ans;
    }
}